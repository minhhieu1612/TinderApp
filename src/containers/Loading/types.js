const LOADING = 'LOADING';
const UNLOAD = 'UNLOAD';

export default {LOADING, UNLOAD};
