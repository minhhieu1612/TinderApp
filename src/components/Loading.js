import React from 'react';
import {ActivityIndicator} from 'react-native';

export default function Loading(props) {
  return <ActivityIndicator {...props} />;
}
