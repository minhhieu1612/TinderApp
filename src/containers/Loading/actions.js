import types from './types.js';

const loading = () => ({type: types.LOADING});
const unload = () => ({type: types.UNLOAD});

export {loading, unload};
