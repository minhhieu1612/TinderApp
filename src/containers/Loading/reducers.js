import types from './types';
const initialState = false;

const myReducer = (state = initialState, action) => {
  switch (action.type) {
    case types.LOADING:
      state = true;
      break;
    case types.UNLOAD:
      state = false;
      break;
    default:
      state = false;
  }
  return state;
};

export default myReducer;
