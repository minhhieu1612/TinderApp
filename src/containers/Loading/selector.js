import React from 'react';
import Loading from '../../components/Loading';
import {connect} from 'react-redux';

const LoadingContainer = (props) => {
  return <Loading {...props} />;
};

const mapStateToProps = null;
const mapDispatchToProps = null;

export default connect(mapStateToProps, mapDispatchToProps)(LoadingContainer);
