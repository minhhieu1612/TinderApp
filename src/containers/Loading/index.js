import reducers from './reducers';
import LoadingContainer from './selector';

export {reducers, LoadingContainer};
