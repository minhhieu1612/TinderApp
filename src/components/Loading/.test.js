import React from 'react';
import renderer from 'react-test-renderer';
import Loading from './index';

it('test Component Loading', () => {
  expect(renderer.create(<Loading />)).toMatchSnapshot();
});
