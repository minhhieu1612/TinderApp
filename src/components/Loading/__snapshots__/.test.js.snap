// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Component Loading 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="small"
/>
`;
